hooks/s3_package.py

```diff
@@ -67,10 +68,11 @@ def run(self):
             )
         )
 
-        # just need return code from make command
-        retcode = os.system("make -C {} &> /dev/null".format(fn_root_dir))
+        p = subprocess.Popen(["make -C {} &> /dev/null".format(fn_root_dir)], bufsize=2048, shell=True, stderr=subprocess.PIPE, close_fds=True)
```

Comment:
It would be better to control stream redirects using python build-in `DEVNULL` to redirect stdout/stderr to `/dev/null`.

```python
p = subprocess.Popen([command], shell = True, stdout = DEVNULL, stderr = DEVNULL)
```

Symbol `DEVNULL` exists in Py3, while Py2 still must simulate it as:

```python
try:
    from subprocess import DEVNULL
except ImportError:
    DEVNULL = open(os.devnull, 'wb')
```

Also would be good to define command variable and reuse it in logger and in subprocess. 

```python
command = 'make -C {}'.format(fn_root_dir)

self.logger.info(
    "Making dependencies with '{}' command, output hidden.".format(command)
)

p = subprocess.Popen(...) // see above
```